lib/server-manager.ts

```diff
@@ -111,8 +113,8 @@ export class ServerManager {
     }
   }
 
-  public getActiveServers(): ActiveServer[] {
-    return this._activeServers.slice()
+  public getActiveServers(): Readonly<ActiveServer[]> {
+    return this._activeServers
```

Comment:
I don't think doing slice helps here in any way here. If someone writes bad code which modifies the server, it is their mistake, and the code is still malfunctioning. Throwing a `slice` on an inherently unsafe architecture doesn't change anything. In interpreted languages like JavaScript, we should rely on the types, docs, etc since the runtime doesn't provide anything.